cmake/CMakeLists.txt

```diff
@@ -73,7 +73,7 @@ option(onnxruntime_ENABLE_MICROSOFT_INTERNAL "Use this option to enable/disable
 option(onnxruntime_USE_NUPHAR "Build with Nuphar" OFF)
 option(onnxruntime_USE_BRAINSLICE "Build with BrainSlice" OFF)
 option(onnxruntime_USE_TENSORRT "Build with TensorRT support" OFF)
-option(onnxruntime_ENABLE_LTO "Enable link time optimization" ON)
+option(onnxruntime_ENABLE_LTO "Enable link time optimization" OFF)
```

Comment:
Why default to OFF? Wouldn't this option provide performance benefits?

Reply:
At least it is good for binary size. I saw it reduced binary size from 8MB to 6.1 MB on Linux, with Release build config. Release is bigger than RelWithDebInfo because it enables inlining. 

However, for gcc it's experimental, for older GCCs it could cause some undefined behaviors and at least we are sure it doesn't work well on Ubuntu 16.04.